r2_localization:
  frequency: 50
  sensor_timeout: 0.1
  two_d_mode: true
  
#  map_frame: r2/odom
  odom_frame: r2/odom
  base_link_frame: r2/base_link
  world_frame: r2/odom
  
  transform_time_offset: 0.0
  transform_timeout: 0.2
  
  odom0: mobile_base_controller/odom
  odom0_config: [false, false, false,
                  false, false, false,
                  true,  true, false,
                  false, false, true,
                  false, false, false]
  odom0_queue_size: 2 # Odom is 50Hz
  odom0_relative: false
  odom0_nodelay: true
  
  imu0: imu/data
  imu0_config: [false, false, false,
                false, false, false,
                false, false, false,
                false, false, true,
                true, false, false]
  imu0_queue_size: 2 # Imu is 50Hz
  imu0_relative: false
  imu0_nodelay: true
  
  initial_state: [0.0, 0.0, 0.0,
                  0.0, 0.0, 0.0,
                  0.0, 0.0, 0.0,
                  0.0, 0.0, 0.0,
                  0.0 ,0.0, 0.0]
                  
  publish_tf: true
  publish_acceleration: false
  print_diagnostics: false
  
  use_control: true
  stamped_control: false
  control_timeout: 0.1 # cmd_vel is 20Hz
  control_config: [true, false, false,
                    false, false, true]
  acceleration_limits: [0.8, 0.0, 0.0,
                        0.0, 0.0, 1.5] # See velocity_smoother_controller
  deceleration_limits: [0.8, 0.0, 0.0,
                        0.0, 0.0, 1.5] # See velocity_smoother_controller
  
  smooth_lagged_data: false
  history_length: false
  
  debug: false
  debug_out_file: r2_filtered_odom_debug.txt
  
  process_noise_covariance: [0.05, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0.05, 0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0.06, 0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0.03, 0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0.03, 0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0.06, 0,     0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0.025, 0,     0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0.025, 0,    0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0.04, 0,    0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0.01, 0,    0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0.01, 0,    0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0.02, 0,    0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0.01, 0,    0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0.01, 0,
                              0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0.015]
                              
  initial_estimate_covariance: [1e-9, 0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    1e-9, 0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    1e-9, 0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    1e-9, 0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    1e-9, 0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    1e-9, 0,    0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    1e-3, 0,    0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    1e-3, 0,    0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    1e-9, 0,     0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    1e-9,  0,     0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    0,     1e-9,  0,     0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     1.0 ,  0,    0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     1e-9, 0,    0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    1e-9, 0,
                                0,    0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    1e-9]

r2_imu_filter_madgwick:
  gain: 0.1
  zeta: 0.0
  mag_bias_x: 0.0
  mag_bias_y: 0.0
  mag_bias_z: 0.0
  orientation_stddev: 0.0
  world_frame: "enu"
  use_mag: false
  use_magnetic_field_msg: false
  fixed_frame: "odom"
  publish_tf: false
  reverse_tf: false
  constant_dt: 0.0
  publish_debug_topics: false
  stateless: false

r2_amcl:
  min_particles: 500
  max_particles: 5000
  kld_err: 0.05
  kld_z: 0.99
  update_min_d: 0.0
  update_min_a: 0.0
  resample_interval: 1
  transform_tolerance: 0.1
  recovery_alpha_slow: 0.001
  recovery_alpha_fast: 0.1
  initial_pose_x: 0.0
  initial_pose_y: 0.0
  initial_pose_a: 0.0
  initial_cov_xx: 100.0 # 1.0²
  initial_cov_yy: 1000.0 # 1.5²
  #initial_cov_aa: 39.47 # (2*pi)²
  gui_publish_rate: 10.0
  save_pose_rate: 0.5
  use_map_topic: true
  first_map_only: false

  laser_min_range: -1.0
  laser_max_range: -1.0
  laser_max_beams: 30
  laser_z_hit: 0.5
  laser_z_short: 0.05
  laser_z_max: 0.05
  laser_z_rand: 0.5
  laser_sigma_hit: 0.2
  laser_lambda_short: 0.1
  laser_likelihood_max_dist: 2.0
  laser_model_type: "likelihood_field"

  do_beamskip: false
  beam_skip_distance: 0.5
  beam_skip_threshold: 0.3
  
  odom_model_type: "diff-corrected"
  odom_alpha5: 0.1
  # Publish scans from best pose at a max of 10 Hz
  odom_alpha1: 0.2
  odom_alpha2: 0.2
  # translation std dev, m
  odom_alpha3: 0.8
  odom_alpha4: 0.2
  #odom_alpha5: 0.2
  odom_frame_id: "r2/odom"
  base_frame_id: "r2/base_link"
  global_frame_id: "map"
  tf_broadcast: true

r2_snap_map:
  odom_frame: "r2/odom"
  laser_frame: "r2/laser_link"
  base_frame: "r2/base_link"
  map_frame: "map"